File: jimmer-example/spring-boot-with-jimmer-multiple-datasource/src/main/java/org/jimmer/jimmerdatasouce/controller/WebController.java
package org.jimmer.jimmerdatasouce.controller;

import org.babyfish.jimmer.sql.JSqlClient;
import org.jimmer.jimmerdatasouce.domain.*;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 多数据源测试
 *
 * @author jhlz
 * @version x.x.x
 */
@RestController
public class WebController {

    final JSqlClient sqlClient;

    public WebController(JSqlClient sqlClient) {
        this.sqlClient = sqlClient;
    }

    @Transactional("tm1")
    @GetMapping("/list")
    public List<Student> getStudent() {
        StudentTable table = StudentTable.$;

        Student student = sqlClient.findById(Student.class, 5);
        System.out.println(student);

        return sqlClient.createQuery(table)
                .select(table.fetch(
                        StudentFetcher.$
                                .allScalarFields()
                                .fullName()
                                .courses(
                                        CourseFetcher.$
                                                .courseName()
                                )
                                .college(
                                        CollegeFetcher.$
                                                .collegeName()
                                )
                ))
                .execute();
    }

    @GetMapping("/author")
    @Transactional("tm2")
    public void getDept() {
        AuthorTable table = AuthorTable.$;
        List<Author> list = sqlClient.createQuery(table)
                .select(
                        table.fetch(
                                AuthorFetcher.$
                                        .allScalarFields()
                        )
                )
                .execute();
        list.forEach(System.out::println);
    }
}

File: jimmer-example/micronaut-with-jimmer-kt/src/main/kotlin/com/example/controller/WebController.kt
package com.example.controller

import com.example.domain.Student
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import io.micronaut.serde.annotation.Serdeable
import org.babyfish.jimmer.sql.kt.KSqlClient

/**
 * @author jhlz
 * @version x.x.x
 */
@Controller("/test")
class WebController(val sqlClient: KSqlClient) {


    @Get("/jimmer")
    fun getSqlClient() {
        val list = sqlClient.createQuery(Student::class) {
            select(table)
        }.execute()
        println(list)
    }

    @Get("/user")
    @Produces(MediaType.APPLICATION_JSON)
    fun getUser() = User("admin", "123");


    @Serdeable
    data class User(val username: String, val password: String)

}